{
  "ConnectionPluginManager.unknownPluginCode": "Unknown plugin code: '%s'",
  "ConnectionPluginManager.failedToConnectWithNewTargetClient": "Failed to connect with a new target for host: '%s'",
  "ConnectionProvider.unsupportedHostInfoSelectorStrategy": "Unsupported host selection strategy '%s' specified for this connection provider '%s'. Please visit the documentation for all supported strategies.",
  "DialectManager.unknownDialectCode": "Unknown dialect code: '%s'.",
  "DialectManager.getDialectError": "Was not able to get a database dialect.",
  "DefaultPlugin.executingMethod": "Executing method: %s",
  "DefaultConnectionPlugin.unknownRoleRequested": "A HostInfo with a role of HostRole.UNKNOWN was requested via getHostInfoByStrategy. The requested role must be either HostRole.WRITER or HostRole.READER",
  "DefaultConnectionPlugin.noHostsAvailable": "The default connection plugin received an empty host list from the plugin service.",
  "HostSelector.noHostsMatchingRole": "No hosts were found matching the requested '%s' role.",
  "IamAuthenticationPlugin.unsupportedHostname": "Unsupported AWS hostname %s. Amazon domain name in format *.AWS-Region.rds.amazonaws.com or *.rds.AWS-Region.amazonaws.com.cn is expected.",
  "IamAuthenticationPlugin.useCachedIamToken": "Use cached IAM token = '%s'",
  "IamAuthenticationPlugin.generatedNewIamToken": "Generated new IAM token = %s",
  "IamAuthenticationPlugin.invalidPort": "Port number: %d is not valid. Port number should be greater than zero. Falling back to default port.",
  "IamAuthenticationPlugin.unhandledException": "Unhandled exception: %s",
  "IamAuthenticationPlugin.connectException": "Error occurred while opening a connection: %s",
  "AwsSecretsManagerConnectionPlugin.failedToFetchDbCredentials": "Was not able to either fetch or read the database credentials from AWS Secrets Manager. Ensure the correct secretId and region properties have been provided.",
  "AwsSecretsManagerConnectionPlugin.missingRequiredConfigParameter": "Configuration parameter 'secretId' is required.",
  "AwsSecretsManagerConnectionPlugin.unhandledException": "Unhandled exception: '%s'",
  "AwsSecretsManagerConnectionPlugin.endpointOverrideInvalidConnection": "A connection to the provided endpoint could not be established: '%s'.",
  "PluginManager.PipelineNone": "A pipeline was requested but the created pipeline evaluated to None.",
  "ClusterAwareReaderFailoverHandler.invalidTopology": "'%s' was called with an invalid (null or empty) topology",
  "ClusterAwareReaderFailoverHandler.attemptingReaderConnection": "Trying to connect to reader: '%s', with properties '%s'",
  "ClusterAwareReaderFailoverHandler.successfulReaderConnection": "Connected to reader: '%s'",
  "ClusterAwareReaderFailoverHandler.failedReaderConnection": "Failed to connect to reader: '%s'",
  "Utils.topology": "Topology: %s",
  "RdsHostListProvider.incorrectDialect": "Dialect needs to be a topology aware dialect.",
  "RdsHostListProvider.suggestedClusterId": "ClusterId '%s' is suggested for url '%s'.",
  "ConnectionStringHostListProvider.parsedListEmpty": "Can't parse connection string: '%s'.",
  "ConnectionStringHostListProvider.errorIdentifyConnection": "An error occurred while obtaining the connection's host ID.",
  "ExecuteTimePlugin.executeTime": "Executed method '%s' in %s milliseconds.",
  "ConnectTimePlugin.connectTime": "Connected to '%s' in %s milliseconds.",
  "PluginManager.failedToConnectWithNewTargetClient": "PluginManager.failedToConnectWithNewTargetClient",
  "ClusterAwareWriterFailoverHandler.failoverCalledWithInvalidTopology": "Failover was called with an invalid (null or empty) topology.",
  "ClusterAwareWriterFailoverHandler.failedToConnectToWriterInstance": "Failed to connect to the writer instance.",
  "ClusterAwareWriterFailoverHandler.successfulConnectionInvalidTopology": "'%s' successfully established a connection but doesn't contain a valid topology.",
  "ClusterAwareWriterFailoverHandler.successfullyConnectedToNewWriterInstance": "Successfully connected to the new writer instance: '%s'",
  "ClusterAwareWriterFailoverHandler.successfullyReconnectedToWriterInstance": "Successfully re-connected to the current writer instance: '%s'",
  "ClusterAwareWriterFailoverHandler.taskAAttemptReconnectToWriterInstance": "[TaskA] Attempting to re-connect to the current writer instance: '%s', with properties '%s'",
  "ClusterAwareWriterFailoverHandler.taskAEncounteredException": "[TaskA] encountered an exception: '%s'",
  "ClusterAwareWriterFailoverHandler.taskAFinished": "[TaskA] Finished",
  "ClusterAwareWriterFailoverHandler.taskBAttemptConnectionToNewWriterInstance": "[TaskB] Attempting to connect to a new writer instance, with properties '%s'",
  "ClusterAwareWriterFailoverHandler.taskBEncounteredException": "[TaskB] encountered an exception: '%s'",
  "ClusterAwareWriterFailoverHandler.taskBFinished": "[TaskB] Finished",
  "ClusterAwareWriterFailoverHandler.taskBConnectedToReader": "[TaskB] Connected to reader: '%s'",
  "ClusterAwareWriterFailoverHandler.taskBFailedToConnectToAnyReader": "[TaskB] Failed to connect to any reader.",
  "ClusterAwareWriterFailoverHandler.standaloneNode": "[TaskB] Host %s is not yet connected to a cluster. The cluster is still being reconfigured.",
  "ClusterAwareWriterFailoverHandler.taskBAttemptConnectionToNewWriter": "[TaskB] Trying to connect to a new writer: '%s'",
  "ClusterAwareWriterFailoverHandler.alreadyWriter": "Current reader connection is actually a new writer connection.",
  "Failover.TransactionResolutionUnknownError": "Transaction resolution unknown. Please re-configure session state if required and try restarting the transaction.",
  "Failover.connectionChangedError": "The active SQL connection has changed due to a connection failure. Please re-configure session state if required.",
  "Failover.parameterValue": "%s = %s",
  "Failover.unableToConnectToWriter": "Unable to establish SQL connection to the writer instance.",
  "Failover.unableToConnectToReader": "Unable to establish SQL connection to the reader instance.",
  "Failover.detectedException": "Detected an exception while executing a command: %s",
  "Failover.failoverDisabled": "Cluster-aware failover is disabled.",
  "Failover.establishedConnection": "Connected to %s",
  "Failover.startWriterFailover": "Starting writer failover procedure.",
  "Failover.startReaderFailover": "Starting reader failover procedure.",
  "Failover.invalidNode": "Node is no longer available in the topology: %s",
  "Failover.noOperationsAfterConnectionClosed": "No operations allowed after cliend ended.",
  "Failover.transactionResolutionUnknownError": "Failover.transactionResolutionUnknownError",
  "StaleDnsHelper.clusterEndpointDns": "Cluster endpoint resolves to '%s'.",
  "StaleDnsHelper.writerHostInfo": "Writer host: '%s'.",
  "StaleDnsHelper.writerInetAddress": "Writer host address: '%s'",
  "StaleDnsHelper.staleDnsDetected": "Stale DNS data detected. Opening a connection to '%s'.",
  "StaleDnsHelper.reset": "Reset stored writer host.",
  "StaleDnsPlugin.requireDynamicProvider": "Dynamic host list provider is required.",
  "Client.methodNotSupported": "Method not supported.",
  "Client.invalidTransactionIsolationLevel": "An invalid transaction isolation level was provided: '%s'.",
  "AdfsCredentialsProviderFactory.failedLogin": "Failed login. Could not obtain SAML Assertion from ADFS SignOn Page POST response: \n '%s'",
  "AdfsCredentialsProviderFactory.getSamlAssertionFailed": "Failed to get SAML Assertion due to exception:  '%s'",
  "AdfsCredentialsProviderFactory.invalidHttpsUrl": "Invalid HTTPS URL: '%s'",
  "AdfsCredentialsProviderFactory.signOnPagePostActionUrl": "ADFS SignOn Action URL: '%s'",
  "AdfsCredentialsProviderFactory.signOnPagePostActionRequestFailed": "ADFS SignOn Page POST action failed with HTTP status '%s', reason phrase '%s', and response '%s'",
  "AdfsCredentialsProviderFactory.signOnPageRequestFailed": "ADFS SignOn Page Request Failed with HTTP status '%s', reason phrase '%s', and response '%s'",
  "AdfsCredentialsProviderFactory.signOnPageUrl": "ADFS SignOn URL: '%s'",
  "FederatedAuthPlugin.generatedNewIamToken": "Generated new IAM token = '%s'",
  "FederatedAuthPlugin.unhandledException": "Unhandled exception: '%s'",
  "FederatedAuthPlugin.unsupportedHostname": "Unsupported AWS hostname '%s'. Amazon domain name in format *.AWS-Region.rds.amazonaws.com or *.rds.AWS-Region.amazonaws.com.cn is expected.",
  "FederatedAuthPlugin.useCachedIamToken": "Use cached IAM token = '%s'",
  "FederatedAuthPluginFactory.unsupportedIdp": "Unsupported Identity Provider '%s'. Please visit to the documentation for supported Identity Providers."
}
